flocker/node/agents/ebs.py

```diff
@@ -128,15 +106,18 @@ def _run_with_logging(self, *args, **kwargs):
         """
         original = getattr(self, original_name)
         method = getattr(original, method_name)
-        try:
-            return method(*args, **kwargs)
-        except EC2ResponseError as e:
-            BOTO_EC2RESPONSE_ERROR(
-                code=e.code,
-                message=e.message,
-                request_id=e.request_id,
-            ).write()
-            raise
+
+        # Trace IBlockDeviceAPI ``method`` as Eliot Action.
+        with AWS_ACTION(operation=[method_name, args, kwargs]):
+            try:
+                return method(*args, **kwargs)
+            except EC2ResponseError as e:
+                BOTO_EC2RESPONSE_ERROR(
+                    aws_code=e.code,
+                    aws_message=e.message,
+                    aws_request_id=e.request_id,
+                ).write()
```

Comment:
> All the information logged in the message is already logged by the failed action...wrapped in XML tags...but the info is there already.

True. 2 reasons why having ``BOTO_EC2RESPONSE_ERROR`` is useful:
1) The error fields are nicely presented for future processing.
2) Easier to unit test using ``test_boto_ec2response_error`` in ``test_ebs.py``.

Please let me know if the above two reasons for having explicit ``BOTO_EC2RESPONSE_ERROR`` do not sound reasonable? Thanks!